bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/RuleUtils.java

```diff
@@ -0,0 +1,373 @@
+import org.slf4j.LoggerFactory;
+
+/**
+ * Rule utility methods. The Hue scheduler and Hue rules support is based on the automation engine.
+ * This class provides methods to convert between Hue entries and automation rules.
+ *
+ * @author David Graeff - Initial contribution
+ */
+@NonNullByDefault
+public class RuleUtils {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RuleUtils.class);
+    public static Random random = new Random(); // public for test mock
+
+    /**
+     * Splits the given base time (pattern "hh:mm[:ss]") on the colons and return the resulting array.
+     * If an upper time is given (same pattern), a random number between base and upper time is chosen.
+     *
+     * @param baseTime A base time with the pattern hh:mm[:ss]
+     * @param upperTime An optional upper time or null or an empty string
+     * @return Time components (hour, minute, seconds).
+     */
+    public static String[] computeRandomizedDayTime(String baseTime, @Nullable String upperTime) {
+        if (!baseTime.matches("[0-9]{1,2}:[0-9]{1,2}(:[0-9]{1,2})?")) {
+            throw new IllegalStateException("Time pattern incorrect. Must be 'hh:mm[:ss]'. " + baseTime);
+        }
+
+        String r[] = baseTime.split(":");
+
+        if (upperTime != null && !upperTime.isEmpty()) {
+            String[] upperTimeParts = upperTime.split(":");
+            if (!upperTime.matches("[0-9]{1,2}:[0-9]{1,2}(:[0-9]{1,2})?") || r.length != upperTimeParts.length) {
+                throw new IllegalStateException("Random Time pattern incorrect. Must be 'hh:mm[:ss]'. " + upperTime);
+            }
+            for (int i = 0; i < r.length; ++i) {
+                int n = Integer.parseInt(r[i]);
+                int n2 = Integer.parseInt(upperTimeParts[i]);
+                int diff = Math.abs(n2 - n); // Example: 12 and 14 -> diff = 2
+                if (diff > 0) { // diff = rnd [0,3)
+                    diff = random.nextInt(diff + 1);
+                }
+                r[i] = String.valueOf(n2 > n ? n + diff : n2 + diff);
+            }
+        }
+
+        return r;
+    }
+
+    /**
+     * Validates a hue http address used in schedules and hue rules.
+     *
+     * @param ds A hue datastore to verify that referred lights/groups do exist
+     * @param address Relative hue API address. Example: "/api/<username>/groups/1/action" or
+     *            "/api/<username>/lights/1/state"
+     * @throws IllegalStateException Thrown if address is invalid
+     */
+    @SuppressWarnings({ "unused", "null" })
+    public static void validateHueHttpAddress(HueDataStore ds, String address) throws IllegalStateException {
+        String[] validation = address.split("/");
+        if (validation.length < 6 || !validation[0].isEmpty() || !validation[1].equals("api")) {
+            throw new IllegalStateException("Given address invalid!");
+        }
+        if (validation[3].equals("groups") && validation[5].equals("action")) {
```

Comment:
Just as a general guideline :-)

Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)